docs/site/gpu.md

```diff
@@ -65,6 +83,84 @@ CUDA toolkit
   4. [CUDNN 7.x](https://developer.nvidia.com/cudnn)
 
+## Linux
+
+## Linux
+
+One easiest way to install the NVIDIA software is with `apt` on Ubuntu. For other distributions
+refer to the [CUDA install Linux](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html).
+
+Note: All linux distributions may not support this. you might encounter some problems with driver
+installations.
+
+To check the CUDA compatible driver version:
+
+```sh
+
+# linux-modules-nvidia-390-5.0.0-1018-azure
+#                                    -aws, -gcp, -oracle, -generic, or -lowlatency
+
+# Get the latest driver version number, significant digit 5 like 435, 450
+NVIDIA_DRIVER_VERSION=$(sudo apt-cache search 'linux-modules-nvidia-[0-9]+-generic$' | awk '{print $1}' | sort | tail -n 1 | head -n 1 | awk -F"-" '{print $4}')
+
+CUDA_DRIVER_VERSION=$(apt-cache madison nvidia-cuda-toolkit | awk '{print $3}' | sort -r | while read line; do
+   if dpkg --compare-versions $(dpkg-query -f='${Version}\n' -W nvidia-driver-${NVIDIA_DRIVER_VERSION}) ge $line ; then
+       echo "$line"
+       break
+   fi
+done)
+```
+
+Install [CUPTI](http://docs.nvidia.com/cuda/cupti/) which ships with CUDA toolkit for profiling.
+
+```sh
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/extras/CUPTI/lib64
+```
+
+### Install CUDA with apt
+
+The following instructions are for installing CUDA 10.2 on Ubuntu 18.04. These instructions
+might work for other Debian-based distros.
+
+Note: [Secure Boot](https://wiki.ubuntu.com/UEFI/SecureBoot) tends to complication installation.
+These instructions may not address this.
+
+#### Ubuntu 18.04 (CUDA 10.2)
+
+```sh
+
+# Add NVIDIA package repositories
+# 1. Download the Ubuntu 18.04 driver repository
+wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-ubuntu1804.pin
+# 2. Move the repository to preferences
+sudo mv cuda-ubuntu1804.pin /etc/apt/preferences.d/cuda-repository-pin-600
+# 3. Fetch keys
+sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/7fa2af80.pub
+# 4. add repository
+sudo add-apt-repository "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/ /"
+# 5. Update package lists
+sudo apt-get update
+
+# ---
+# get the machine-learning repo
+wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb
+
+sudo apt install ./nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb
+sudo apt-get update
+
+wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/libcudnn7_7.6.5.32-1+cuda10.2_amd64.deb
+sudo apt install ./libcudnn7_7.6.5.32-1+cuda10.2_amd64.deb
+sudo apt-get update
+
+# ---
+
+# Install development and runtime libraries (~4GB)
+sudo apt-get install --no-install-recommends \
+    cuda-10-2
+# Reboot the system. And run `nvidia-smi` for GPU check.
```

Comment:
```console
$ nvidia-smi
Thu May 13 04:19:11 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 465.19.01    Driver Version: 465.19.01    CUDA Version: 11.3     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA Tesla K80    Off  | 00000000:00:1E.0 Off |                    0 |
| N/A   38C    P0    58W / 149W |      0MiB / 11441MiB |     98%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

```

Reply:
The sample did not run 

Error:
```console
In file included from cudaNvSci.cpp:12:0:
cudaNvSci.h:14:10: fatal error: nvscibuf.h: No such file or directory
 #include <nvscibuf.h>
          ^~~~~~~~~~~~
compilation terminated.
Makefile:394: recipe for target 'cudaNvSci.o' failed
make[1]: *** [cudaNvSci.o] Error 1
make[1]: Leaving directory '/home/ubuntu/nvidia-cuda-10.2-samples/0_Simple/cudaNvSci'
Makefile:51: recipe for target '0_Simple/cudaNvSci/Makefile.ph_build' failed
make: *** [0_Simple/cudaNvSci/Makefile.ph_build] Error 2

```

Issue https://github.com/NVIDIA/cuda-samples/issues/22

Reply:
Use `make -k` to continue build with skipping error.

Reply:
But sample did not build. 😕